src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -490,7 +494,9 @@ class DecisionTree :
   std::vector<DecisionTree*> children;
   //! The dimension this node splits on.
   size_t splitDimension;
-
+  //! The reference dataset
+  MatType dataset;
```

Comment:
Holding the matrix inside the tree is extremely inefficient as I mentioned earlier.

Reply:
Shall I add a integer variable in replace that will be equal to dimensionality otherwise I can't add the check that training and testing data will have same features.